pkg/registry/core/service/strategy.go

```diff
@@ -185,6 +185,11 @@ func dropServiceDisabledFields(newSvc *api.Service, oldSvc *api.Service) {
 			ing.IPMode = nil
 		}
 	}
+
+	// Clear AllocateLoadBalancerNodePorts if ServiceLBNodePortControl if not enabled
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ServiceLBNodePortControl) {
```

Comment:
This check used to check for `!allocateLoadBalancerNodePortsInUse` but it seems like we dropped it, was that intentional?

Reply:
Yes, since the allocateLoadBalancerNodePorts must be nil if the feature-gate is not set. At least that's how I have understood it.

Reply:
> allocateLoadBalancerNodePorts must be nil if the feature-gate is not set

This and if an existing resource has the field already set. So this check should still check if `oldSvc` was using this field

Reply:
Yeah this needs to preserve the field if it was already set in oldSvc.  The rationale is that a user might upgrade to 1.21, use the field, temporarily roll back to 1.20 (where it is gated) - the field value should be preserved so that when they roll forward it is still present.

If this is the only issue I will approve but we need a followup to fix this

Reply:
Re-skim: This will need a followup, still.